lib/index.js

```diff
@@ -22,6 +22,7 @@ const memorize = require("./util/memorize");
 /** @typedef {import("../declarations/WebpackOptions").WebpackOptionsNormalized} WebpackOptionsNormalized */
 /** @typedef {import("../declarations/WebpackOptions").WebpackPluginFunction} WebpackPluginFunction */
 /** @typedef {import("../declarations/WebpackOptions").WebpackPluginInstance} WebpackPluginInstance */
+/** @typedef {import("../declarations/WebpackOptions").WebpackPluginInstance} Plugin */
```

Comment:
WebpackPluginInstance and Plugin are not the same.
WebpackPluginInstance is an interface, Plugin in `@types/webpack` is an abstract class.
One can't use an interface with `class extends WebpackPluginInstance`.

You can create a new `Plugin` type as `abstract class` to create compatibility.

Reply:
ol. I'll do it 👍

Reply:
Hi sokra and pastak,
i also encountered this problem. Are you going to fix it soon?

Reply:
Hi, I am still facing this issue -- any way there is going to be a fix?